src/ibonwe/shape/diamond.cpp

```diff
@@ -0,0 +1,53 @@
+#include <iostream>
+
+#include "ibonwe/shape/diamond.h"
+
+namespace ibonwe {
+namespace shape {
+
+/**
+ * The default diamond constructor
+ */
+ibonwe::shape::diamond::diamond()
+{
+    character = ' ';
+    width     = 0;
+    height    = 0;
+}
+
+/**
+ * The parameterised diamond constructor
+ *
+ * @param char aCharacter A diamond character
+ * @param int  aHeight    A diamond height
+ */
+ibonwe::shape::diamond::diamond(const char aCharacter, const int aHeight)
+{
+    character = aCharacter;
+    width     = aHeight;
+    height    = aHeight;
+}
+
+/**
+ * The purpose of this method is to draw a diamond
+ */
+void ibonwe::shape::diamond::draw()
+{
+    for (int i = 0; i < (height * 0.5); i++) {
```

Comment:
I think it would be much easier on the eye to read divide by two than multiple by 0.5.